service/matching/matching_engine.go

```diff
@@ -199,7 +206,7 @@ func NewEngine(
 		visibilityManager:         visibilityManager,
 		metricsHandler:            metricsHandler.WithTags(metrics.OperationTag(metrics.MatchingEngineScope)),
 		partitions:                make(map[tqid.PartitionKey]taskQueuePartitionManager),
-		taskQueueCount:            make(map[taskQueueCounterKey]int),
+		gaugeMetricCounter:        gaugeMetrics{make(map[taskQueueCounterKey]int), make(map[taskQueueCounterKey]int), make(map[taskQueueCounterKey]int), make(map[taskQueueCounterKey]int), sync.Mutex{}},
```

Comment:
use key names in initializers and split this on multiple lines. also you don't need to initialize mutex
```suggestion
		gaugeMetricCounter:        gaugeMetrics{
			loadedTaskQueueFamilyCount: make(map[taskQueueCounterKey]int),
			loadedTaskQueueCount: make(map[taskQueueCounterKey]int),
			loadedTaskQueuePartitionCount: make(map[taskQueueCounterKey]int),
			loadedPhysicalTaskQueueCount: make(map[taskQueueCounterKey]int),
		},
```